jerry-core/lit/lit-strings.h

```diff
@@ -135,4 +135,6 @@ ecma_char_t lit_utf8_peek_prev (const lit_utf8_byte_t *);
 void lit_utf8_incr (const lit_utf8_byte_t **);
 void lit_utf8_decr (const lit_utf8_byte_t **);
 
+bool lit_utf8_contains_four_bytes_unicode (const lit_utf8_byte_t *, lit_utf8_size_t);
```

Comment:
I liked the original approach that the size is also computed by this function, and it can be passed to ecma_new_ecma_string_from_utf8_converted_to_cesu8 to avoid double scan.